server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -148,6 +152,12 @@ public class Node implements Closeable {
         Property.NodeScope
     );
 
+    public static final Setting<TimeValue> SHUTDOWN_SEARCH_TIMEOUT_SETTING = Setting.positiveTimeSetting(
+        "node.async_search_grace_period",
```

Comment:
nit: can we use the already existing async_search prefix? eg `async_search.shutdown_grace_period`. While I understand why it's difficult to move this shutdown check into the async search module right now, I want the setting name to be appropriate for eventually moving into there.